hardware_interface/include/hardware_interface/actuator_hardware_interface.hpp

```diff
@@ -0,0 +1,108 @@
+// Copyright 2020 ros2_control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HARDWARE_INTERFACE__ACTUATOR_HARDWARE_INTERFACE_HPP_
+#define HARDWARE_INTERFACE__ACTUATOR_HARDWARE_INTERFACE_HPP_
+
+#include <algorithm>
+#include <array>
+#include <string>
+#include <vector>
+
+#include "hardware_interface/hardware_and_component_info.hpp"
+#include "hardware_interface/joint.hpp"
+#include "hardware_interface/types/hardware_interface_return_values.hpp"
+#include "hardware_interface/types/hardware_interface_status_values.hpp"
+#include "hardware_interface/visibility_control.h"
+
+namespace hardware_interface
+{
+
+/**
+  * \brief Virtual Class to implement when integrating a 1 DoF actuator into ros2_control.
+  * The typical examples are conveyors or motors.
+  */
+class ActuatorHardwareInterface
+{
+public:
+  HARDWARE_INTERFACE_PUBLIC
+  ActuatorHardwareInterface() = default;
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  ~ActuatorHardwareInterface() = default;
+
+  /**
+   * @brief Configuration of the actuator from data parsed from the robot's URDF.
+   *
+   * @param actuator_info structure with data from URDF.
+   * @return return_type::OK if required data are provided and can be parsed,
+   * return_type::ERROR otherwise.
+   */
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  return_type configure(const HardwareInfo & actuator_info) = 0;
```

Comment:
Why would this get hardware info opposite to component info?

Reply:
The Components are "Joint" and "Sensor" classes, all other are hardware.

Difference comes from the URDF structure and HardwareInfo holds data about Components. (One could do also third structure, but I do not think this will then look nicer)

Reply:
Doxygen-string explains this now

Reply:
Understood. I believe this has to be made clearer, in form of doxygen at the the structs as well as in a design doc. I can see how this is becoming confusing for people out there.